### PR TITLE
Openvino ep 2022.2 v4.2

### PR DESCRIPTION
**Pull request by sfatimar**
This changes are to align OV 2022.2 Release with ORT . Changes
CPU FP16 Support, dGPU Support, RHEL Dockerfile, Ubuntu 20 Dockerfile 

**Motivation and Context**
- This change is required to ensure ORT-OpenVINO Execution Provider is aligned with latest changes. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: dockerfiles/Dockerfile.openvino-rhel8 [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Build stage
+FROM registry.access.redhat.com/ubi8/ubi:8.4
```
Comment (snnn) [inline dockerfiles/Dockerfile.openvino-rhel8:2]: According to the discussions in https://github.com/pypa/manylinux/issues/1282, I don't think taking dependency on UBI8 is good. Especially, licenses. 
When you add extra yum repos to this dockerfile, it makes the built images non-distributable.  See: "https://developers.redhat.com/articles/ubi-faq#ubi_details" . "Can I add non-UBI RPMs to a UBI image and still redistribute the resultant container image on a non-Red Hat platform?" If I understand correctly, the answer is no. And you added CentOS 8 stream packages to this image too. Then why don't you use a centos 8 stream base image instead? 

Red hat people says:"CentOS Stream 8 is the representation of RHEL 8 Nightly". We don't use nightly packages in productions. Therefore I suggest we should avoid using CentOS Stream 8 too.

Comment (sfatimar) [reply dockerfiles/Dockerfile.openvino-rhel8:2]: Do you have any suggestions what we can use

Comment (snnn) [reply dockerfiles/Dockerfile.openvino-rhel8:2]: What's the purpose of this docker file? Files in this dir are not for build pipelines. We put pipelines' docker files in https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/linux/docker. So, if the packages built from this docker file won't be distributed separately, you may just ubuntu. Any Ubuntu version is fine. If you want to continue to use ubi8, you should avoid using packages from CentOS 8 stream and be careful on license restrictions.

Comment (sfatimar) [reply dockerfiles/Dockerfile.openvino-rhel8:2]: This is not part of pipeline but standalone script we maintain to ensure we can build RHEL Docker Image for some customers . We have removed centos8 elements , is this ok to go now.

**Comment by sfatimar** [comment]
@snnn Please confirm on RHEL Dockerfile

**Comment by snnn** [comment]
> @snnn Please confirm on RHEL Dockerfile

The content LGTM. But, why is it in ORT repository while it is only for providing a support for some customers that I don't know who they are and what they need? We can't help you maintain this file, and the file doesn't have any document with build instructions.  The GCC version in UBI8 is 8.x. We may deprecate the support of GCC 8.x at anytime. I'm discussing it with the other developers in ORT team. Right now no conclusion is made yet. So I can't promise we will list UBI8 as a supported building environment. As we are approaching the deadline of cutting of a release branch, we can't rush to make decisions.

**Comment by snnn** [comment]
I'm trying to understand why these docker files are in this repo and how they are used.

**Review by jywu-msft — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Comment by jywu-msft** [comment]
> 

i think maybe we should move these dockerfiles somewhere else?  inference-examples perhaps?

**Comment by sfatimar** [comment]
i think maybe we should move these dockerfiles somewhere else? inference-examples perhaps?

Sure George lets discuss this in our next meeting. I think we should be able to host it in Intel Repo somewhere

**Pull request merged**